### PR TITLE
feat: pktmon client recreation + remove ltsc2019 in azp

### PR DESCRIPTION
**Pull request by matmerr**
# Description
Occasionally the grpc stream experiences transient errors, which are mitigated by recreating the client observer stream. Requires further RCA, but this avoids crashloopbackoff, and gracefully logs to continue

## Related Issue

If this pull request is related to any issue, please mention it here. Additionally, make sure that the issue is assigned to you before submitting this pull request.

## Checklist

- [ ] I have read the [contributing documentation](https://retina.sh/docs/contributing).
- [ ] I signed and signed-off the commits (`git commit -S -s ...`). See [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) on signing commits.
- [ ] I have correctly attributed the author(s) of the code.
- [ ] I have tested the changes locally.
- [ ] I have followed the project's style guidelines.
- [ ] I have updated the documentation, if necessary.
- [ ] I have added tests, if applicable.

## Screenshots (if applicable) or Testing Completed

Please add any relevant screenshots or GIFs to showcase the changes made.

## Additional Notes

Add any additional notes or context about the pull request here.

---

Please refer to the [CONTRIBUTING.md](../CONTRIBUTING.md) file for more information on how to contribute to this project.

### CODE REVIEW

File: pkg/plugin/windows/pktmon/pktmon_plugin_windows.go [hunk 1]
```diff
@@ -129,6 +130,44 @@ func (p *Plugin) Start(ctx context.Context) error {
...
+			if err != nil {
+				p.l.Error("failed to get flow from observer", zap.Error(err))
+				// seeing status errors, recreate client
+				// experiencing sporadic cases of this
+
+				// and then client gets stuck in loop, where crashing and recreating is immediate mitigation.
+				// Instead, we will try to recreate the client and stream
+				if _, ok := status.FromError(err); ok {
+					var clientErr error
+					clientErr = str.CloseSend()
```
Comment (anubhabMajumdar) [inline pkg/plugin/windows/pktmon/pktmon_plugin_windows.go:154]: Is this strictly required? If current client instance is malfunctioning and this fails, plugin bails and takes down entire agent, and new agent will basically perform L158 to mitigate.

Comment (matmerr) [reply pkg/plugin/windows/pktmon/pktmon_plugin_windows.go]: not required, effectively a noop since this a recv stream not send stream, removing

**Review by timraymond — COMMENTED** [review]
I feel like `StartStreamingRPC` isn't doing enough which is making the `err != nil` case do too much. If we're writing a client to deal with getting flows from gRPC, then the user of that shouldn't have to deal with all of gRPC's baggage. There's also spread-out knowledge between `plugin` and `GRPCClient` about how to deal with flows and present them as something useful to the rest of the application.

I think I'd do something like put the contents of `GetFlow` into `StartStreamingRPC` (and rename it `GetFlow` since that's what it does). Then make that return a `<-chan Event` and an `error` and take a `context`. The deal with the caller is that you get things sent on that channel until you cancel the context (at which point the channel is closed).

The idea would really be to make the client a little more specialized for our use case to make using it easier (and less error-prone).

File: pkg/plugin/windows/pktmon/pktmon_plugin_windows.go [hunk 2]
```diff
@@ -111,103 +116,135 @@ func (p *Plugin) RunPktMonServer() error {
...
+	defer cancel()
+
+	err := p.StartStream(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to setup pktmon stream: %w", err)
 	}
 
-	// Write the event to the external channel.
-	if p.externalChannel != nil {
+	for {
```
Comment (huntergregory) [inline pkg/plugin/windows/pktmon/pktmon_plugin_windows.go:225]: It seems like GetFlow functionality has changed? This infinite loop seems new, and the function now keeps getting flows until one of them is nil or an error is reached?

File: pkg/plugin/windows/pktmon/pktmon_plugin_windows.go [hunk 3]
```diff
@@ -111,103 +116,135 @@ func (p *Plugin) RunPktMonServer() error {
...
+						metrics.LostEventsCounter.WithLabelValues(utils.ExternalChannel, string(Name)).Inc()
+					}
+				}
+			}
+		}
 	}()
 
-	var str observerv1.Observer_GetFlowsClient
+	// run the getflows loop
+	for {
```
Comment (huntergregory) [inline pkg/plugin/windows/pktmon/pktmon_plugin_windows.go:166]: Here’s the second infinite loop for getting flows. Remove one of the infinite loops or rename GetFlow to represent the new logic?

**Review by huntergregory — COMMENTED** [review]
Just curious about a new infinite loop

File: pkg/plugin/windows/pktmon/pktmon_plugin_windows.go (hunk 3 above)
Comment (matmerr) [reply pkg/plugin/windows/pktmon/pktmon_plugin_windows.go:166]: the idea here is that p.GetFlow contains a child context, which is represents the lifetime of the streaming rpc. when that rpc bails due to an rpc error, that child context is canceled, and returns here to the caller. if it is an rpc error (which happens a decent amount on windows), restart the streaming rpc via calling GetFlows again. if it's not a known rpc (status) error, bail out of the plugin and crash

File: pkg/plugin/windows/pktmon/pktmon_plugin_windows.go (hunk 2 above)
Comment (matmerr) [reply pkg/plugin/windows/pktmon/pktmon_plugin_windows.go:225]: yep exactly, all of the rpc stream management is here, and parent pktmon plugin is no longer responsible for grpc stream management

File: pkg/plugin/windows/pktmon/pktmon_plugin_windows.go [hunk 4]
```diff
@@ -111,103 +116,135 @@ func (p *Plugin) RunPktMonServer() error {
...
+			if fl == nil {
+				p.l.Error("received nil flow, flow proto mismatch from client/server?")
+				return nil
+			}
+
+			ev := &v1.Event{
+				Event:     fl,
+				Timestamp: fl.GetTime(),
+			}
+			eventCh <- ev
```
Comment (anubhabMajumdar) [inline pkg/plugin/windows/pktmon/pktmon_plugin_windows.go:245]: Why can't we send this to `externalChannel` or `enricher` directly?

File: pkg/plugin/windows/pktmon/pktmon_plugin_windows.go (hunk 3 above)
Comment (huntergregory) [reply pkg/plugin/windows/pktmon/pktmon_plugin_windows.go:137]: Makes sense. I feel like adding a couple comments and renaming GetFlow (e.g. make it plural) might help others who read the code later, but I wouldn’t block the PR on it

**Review by anubhabMajumdar — APPROVED** [review]

**Pull request merged**